contrib/server/operations/pbench-upload-results.py

```diff
@@ -0,0 +1,270 @@
+    )
+    parser.add_argument("-b", "--before", help="Select tarballs older than this")
+    parser.add_argument(
+        "-c",
+        "--checkpoint",
+        type=Path,
+        dest="checkpoint",
+        help="Checkpoint file for restart",
+    )
+    parser.add_argument(
+        "-m",
+        "--metadata",
+        action="append",
+        dest="metadata",
+        help="Set metadata on dataset upload",
+    )
+    parser.add_argument(
+        "-p",
+        "--preview",
+        action="store_true",
+        help="Report actions but make no changes",
+    )
+    parser.add_argument(
+        "-s", "--since", action="store", help="Select tarballs no older than this"
+    )
+    parser.add_argument(
+        "-t", "--token", action="store", dest="token", help="Pbench Server API token"
+    )
+    parser.add_argument(
+        "--verify", "-v", dest="verify", action="store_true", help="Show progress"
+    )
+    parsed = parser.parse_args()
+
+    # Get basic configuration
+    host = socket.gethostname()
+    v = Path("/opt/pbench-server/VERSION")
+    s = Path("/opt/pbench-server/SHA1")
+    version = v.read_text().strip() if v.exists() else "unknown"
+    sha1 = s.read_text().strip() if s.exists() else "unknown"
+    metadata = [
+        f"global.server.legacy.hostname:{host}",
+        f"global.server.legacy.sha1:{sha1}",
+        f"global.server.legacy.version:{version}",
+    ]
+    if parsed.metadata:
+        metadata.extend(parsed.metadata)
+
+    since: Optional[datetime.datetime] = None
+    before: Optional[datetime.datetime] = None
+
+    since_ts: Optional[float] = None
+    before_ts: Optional[float] = None
+
+    if parsed.since:
+        since = date_parser.parse(parsed.since)
+        since_ts = since.timestamp()
+
+    if parsed.before:
+        before = date_parser.parse(parsed.before)
+        before_ts = before.timestamp()
+
+    if since and before:
+        when = f" (from {since:%Y-%m-%d %H:%M} to {before:%Y-%m-%d %H:%M})"
+    elif since:
+        when = f" (since {since:%Y-%m-%d %H:%M})"
+    elif before:
+        when = f" (before {before:%Y-%m-%d %H:%M})"
+    else:
+        when = ""
+
+    if parsed.tarball.is_dir():
+        what = f"DIRECTORY {parsed.tarball}"
+    else:
+        what = f"TARBALL {parsed.tarball}"
+
+    print(f"{what}{when} -> SERVER {parsed.server}")
+
+    checkpoint: Optional[Path] = None
+    processed: list[str] = []
+    if parsed.checkpoint:
+        checkpoint = parsed.checkpoint
+        if parsed.verify:
+            print(f"Processing checkpoint state from {checkpoint}...")
+        if checkpoint.exists():
+            processed = checkpoint.read_text().splitlines()
+            if parsed.verify:
+                print(f"[CPT] done {len(processed)}")
+        if parsed.verify:
+            print(
+                f"Finished processing checkpoint data: {len(processed)} checkpointed files"
+            )
```

Comment:
Does this process really take long enough to justify a "start" and, more significantly, a "finished" message (particularly in that the "finished" message looks pretty redundant with the message at line 177).

It really seems like, if the checkpoint file doesn't exist (or, even if it exists but is empty), then there's not much to do or say here other than, perhaps, to acknowledge the fact that we're gonna keep a checkpoint.  (And, for what it's worth, `CPT` is not a good abbreviation for "checkpoint".)

Reply:
I was going for useful debugging information, not verbose "customer facing" information. If I were to decide to take your suggestion to invert that, I'd consider cleaning up the messages; but that may also weigh against such a decision. 😁